source/Calamari.Tests/KubernetesFixtures/InstallTools.cs

```diff
@@ -218,6 +219,45 @@ public async Task InstallGCloud()
                         return GetGcloudExecutablePath(destinationDirectoryName);
                     });
             }
+            
+            if (CalamariEnvironment.IsRunningOnWindows)
+            {
+                InstallGkeAuthPlugin();
+            }
+        }
+        
+        void InstallGkeAuthPlugin()
+        {
+            var variables = new Dictionary<string, string>();
+            var pythonCopyPath = ExecuteCommandAndReturnResult(GcloudExecutable, "components copy-bundled-python", ".", variables);
+            variables.Add("CLOUDSDK_PYTHON", pythonCopyPath);
+
+            var gkeComponent = ExecuteCommandAndReturnResult($"\"{GcloudExecutable}\"", "components list --filter=\"Name=gke-gcloud-auth-plugin\" --format=\"json\"", ".", variables);
+            var gkeComponentJObject = JArray.Parse(gkeComponent).First();
+            var installedState = gkeComponentJObject["state"]["name"].Value<string>();
+            if (installedState != "Installed")
+            {
+                ExecuteCommandAndReturnResult(GcloudExecutable, "components install gke-gcloud-auth-plugin --quiet", ".", variables);
+            }
+        }
+
+        string ExecuteCommandAndReturnResult(string executable, string arguments, string workingDirectory, Dictionary<string, string> variables)
+        {
+            var stdOut = new StringBuilder();
+            var stdError = new StringBuilder();
+            var commandExitCode = SilentProcessRunner.ExecuteCommand(executable,
+                                                                     arguments,
+                                                                     workingDirectory,
+                                                                     variables,
+                                                                     (Action<string>)(s => stdOut.AppendLine(s)),
+                                                                     (Action<string>)(s => stdError.AppendLine(s))).ExitCode;
+
+            if (commandExitCode != 0)
+            {
+                throw new InvalidOperationException($"stdOut: {stdOut}, stdError: {stdError}");
+            }
+
+            return stdOut.ToString().Trim('\r', '\n');
         }
```

Comment:
I was reading other parts of the same file, looks like this pattern has been done multiple times elsewhere, such as in `InstallAwsCli` and `AddExecutePermission`. Maybe we can make all of them use this helper methods so it is more consistent?

Reply:
yeah, I can fix that up 👍

Reply:
@YihaoOct I made those changes as suggested in [c4420aa](https://github.com/OctopusDeploy/Calamari/pull/971/commits/c4420aabc791995df618372379b69cc53dbb561f)